std/algorithm/iteration.d

```diff
@@ -303,8 +304,11 @@ private struct _Cache(R, bool bidir)
 
     this(R range)
     {
-        source = range;
-        if (!range.empty)
+        static if (__traits(isPOD, R))
```

Comment:
I don't see the point of this since the constructor still receives a copy of the range.

Reply:
If an rvalue is passed to a by-value parameter, it is moved, not copied. So this change allows rvalues of non-copyable types to be passed to this constructor.